tests/conftest.py

```diff
@@ -531,7 +530,9 @@ def fanouthosts(ansible_adhoc, conn_graph_facts, creds, duthosts):      # noqa F
     # WA for virtual testbed which has no fanout
     try:
         for dut_host, value in dev_conn.items():
-            duthost = duthosts[dut_host]
+            # In Python2, the key in dev_conn.items() is unicode object, but in Python3 is AnsileUnsafeText.
+            # So that convert to str explicitly.
+            duthost = duthosts[str(dut_host)]
```

Comment:
Other test scripts using this fixture also may run into such issue. Suggest to fix this issue from `ansible/library/conn_graph_facts.py`.

Reply:
Thanks for the good suggestion! But I tried to modify the `ansible/library/conn_graph_facts.py` like rename the key `device_conn`, but it didn't work:
```
            results = {
                'device_info': lab_graph.devices,
                'device_conn': lab_graph.links,
                'device_port_vlans': lab_graph.vlanport,
            }
```
Currently, I found `conn_graph_facts` was get from here in `tests/common/fixtures/conn_graph_facts.py`:
```
                conn_graph_facts = localhost.conn_graph_facts(
                    **kargs)["ansible_facts"]
```
So, I added a method to make some converts after this `conn_graph_fact` was got if using Python3. This can also fix this issue.
```
def key_convert2str(conn_graph_facts):
    """
        In Python2, some key type are unicode, but In Python3, are AnsibleUnsafeText. Convert them to str.
        Currently, convert conn_graph_facts['device_conn'].
    """
    # If Python2, do not change
    if sys.version_info[0] < 3:
        return conn_graph_facts

    # Else, convert
    result = copy.deepcopy(conn_graph_facts)
    for key, value in conn_graph_facts['device_conn'].items():
        result['device_conn'][str(key)] = result['device_conn'].pop(str(key))

    return result
```